src/azure-cli-core/azure/cli/core/parser.py

```diff
@@ -144,8 +147,22 @@ def _check_value(self, action, value):
         # Override to customize the error message when a argument is not among the available choices
         # converted value must be one of the choices (if specified)
         if action.choices is not None and value not in action.choices:
-            msg = 'invalid choice: {}'.format(value)
-            raise argparse.ArgumentError(action, msg)
+            error_msg = "{prog}: '{value}' is not an {prog} command. See '{prog} --help'.".format(prog=self.prog,
+                                                                                                  value=value)
+            telemetry.set_user_fault(error_msg)
```

Comment:
Why are we sending telemetry when we didn't before?

Reply:
Originally, `_check_value` raises `argparse.ArgumentError` when there is an error, which called [`error`](https://github.com/chewong/azure-cli/blob/bc1347bc6554a8e6fdcfed609a0e7bac1529436a/src/azure-cli-core/azure/cli/core/parser.py#L119-L124) in `parser.py` right after. Since `_check_value` no longer raises `argparse.ArgumentError`, I have to manually send telemetry and exit with exit code 2, which were excuted in [`error`](https://github.com/chewong/azure-cli/blob/bc1347bc6554a8e6fdcfed609a0e7bac1529436a/src/azure-cli-core/azure/cli/core/parser.py#L119-L124)

Reply:
So `error` is no longer called?

Reply:
Yes that's correct